source/acquaman/AMXRFScanController.cpp

```diff
@@ -0,0 +1,159 @@
+#include "AMXRFScanController.h"
+
+#include "analysis/AMnDDeadTimeAB.h"
+#include "analysis/AMAdditionAB.h"
+#include "beamline/AMBeamline.h"
+#include "dataman/AMScan.h"
+#include "dataman/AMUser.h"
+#include "dataman/datastore/AMCDFDataStore.h"
+
+#include <QStringBuilder>
+
+AMXRFScanController::AMXRFScanController(AMXRFScanConfiguration *configuration, QObject *parent)
+	: AMScanController(configuration, parent)
+{
+	detector_ = qobject_cast<AMXRFDetector *>(AMBeamline::bl()->exposedDetectorByInfo(configuration->detectorConfigurations().at(0)));
+
+	scan_ = new AMScan;
+	scan_->setScanConfiguration(configuration);
+	scan_->setName(QString("XRF Scan - %1").arg(detector_->name()));
+	scan_->setFileFormat("amCDFv1");
+	scan_->setFilePath(AMUserSettings::defaultRelativePathForScan(QDateTime::currentDateTime()) % ".cdf");
+	scan_->setRunId(AMUser::user()->currentRunId());
+	scan_->setNotes(QString("%1\n\n%2\n\n").arg(detector_->details()).arg(AMBeamline::bl()->details()));
```

Comment:
Now that the details() mechanism exists, is there a potential to do a similar thing for the other generic scan types?